plugins/dicom_viewer/server/__init__.py

```diff
@@ -134,6 +134,15 @@ def _removeUniqueMetadata(dicomMeta, additionalMeta):
 
 
 def _coerceValue(value):
+    # For binary data, see if it can be coerced further into utf8 data.  If
+    # not, mongo won't store it, so don't accept it here.
+    if isinstance(value, six.binary_type):
```

Comment:
Are these values always coming in as `six.binary_type` or could they be other things, e.g. `bytearray`?

Reply:
I don't know, but we reject things that aren't subclasses of `six.binary_type` or other known types below and raise a `ValueError`.

Reply:
We can do that, which will also need to add `bytearray` to the `knownBaseType` set below.  If you have a `bytearray` as a value, does it insert into Mongo?  Must it be unicode like a `six.binary_type`?

Reply:
Looking at pydicom, it uses `bytearray` for pixel data and some other data.  I'd say since we are many extracting data for use in searches, extracting `bytearray` or true binary data is not particularly desirable.  Even though we could do so, I'm going to recommend we don't (or at least not as part of this PR).  This PR really just tries to get back to the functionality we had of displaying DICOM files regardless of what metadata was in them; the failure to parse binary strings was one  such issue.

Reply:
Fair enough